labeler/input_processor.py

```diff
@@ -71,12 +90,19 @@ def read_data(
                 attention_masks.append(tokenized_input["attention_mask"])
                 token_type_ids.append(tokenized_input["token_type_ids"])
 
+        logging.info("Tokenizing input…")
+        with concurrent.futures.ThreadPoolExecutor() as executor:
```

Comment:
I suspect ProcessPoolExecutor works better since the tokenization is CPU limited

Reply:
Re: your overall point about https://www.tensorflow.org/api_docs/python/tf/keras/utils/text_dataset_from_directory , there is no direct equivalent but if I understood right the Dataset and "DataCollator" entities (which can wrap the tokenisation) could be used more efficiently. Given the loose typing in these libraries and the fact that the RAM needed for training didn't turn out to be that huge, I kept it to this naive solution.

Reply:
Addressed the ProcessPoolExecutor related feedback.

Reply:
Something about the tokenization didn't appreciate being executed in a process pool:

```
    @classmethod
    def dumps(cls, obj, protocol=None):
        buf = io.BytesIO()
>       cls(buf, protocol).dump(obj)
E       AttributeError: Can't pickle local object 'InputProcessor.read_data.<locals>.tokenize_text'
```

... swapped back to thread pool executor for now.